yellowbrick/contrib/prepredict.py

```diff
@@ -0,0 +1,104 @@
+#
+# Author:  Benjamin Bengfort <benjamin@bengfort.com>
+# Created: Mon Jul 12 07:07:33 2021 -0400
+#
+# ID: prepredict.py [] benjamin@bengfort.com $
+
+"""
+PrePredict estimator allows Yellowbrick to work with results produced by an estimator
+prior to the visual diagnostic workflow, particularly for inferences that require
+extensive time or compute resources.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import pathlib
+import numpy as np
+
+from sklearn.base import BaseEstimator
+from sklearn.metrics import accuracy_score, r2_score, silhouette_score
+from yellowbrick.contrib.wrapper import CLASSIFIER, CLUSTERER, REGRESSOR
+
+
+class PrePredict(BaseEstimator):
+    """
+    The Passthrough estimator allows users to specify pre-predicted results to
+    Yellowbrick without the need to input the original estimator. Note that Yellowbrick
+    often uses the learned attributes of the estimator to produce rich visual
+    diagnostics, so this estimator may not work for all Yellowbrick visualizers.
+
+    The passthrough estimator can accept data either in memory as a numpy array or it
+    can accept a string, which it interprets as a path on disk to load the data from.
+
+    Currently passthrough does not support predict_proba or decision_function methods,
+    which it could if it was passed predicted data as 2D array instead of a 1D array.
+
+    Parameters
+    ----------
+    data : array-like, func, or file-like object, string, or pathlib.Path
+        The predicted values wrapped by the estimator and returned on predict() and
+        used by the score function. The default expectation is that data is a 1D numpy
+        array of y_hat or y_pred values produced by some other estimator. Data can also
+        be a func, which is called and returned, or a file-like object, string, or
+        pathlib.Path at which point the data is loaded from disk using ``np.load``.
+
+    estimator_type : str, optional
+        One of "classifier", "regressor", "clusterer", "DensityEstimator", or
+        "outlier_detector" that allows the contrib estimator to pass the scikit-learn
+        ``is_classifier``, etc. functions. If not specified, the Yellowbrick visualizer
+        you're trying to use may error.
+    """
+
+    def __init__(self, data, estimator_type=None):
+        self.data = data
+        self._estimator_type = estimator_type
+
+    def fit(self, X, y=None):
+        """
+        Fit is a no-op, simply returning self per the scikit-learn API.
+        """
+        return self
+
+    def predict(self, X):
+        """
+        Predict returns the embedded data but does not perform any checks on the
+        validity of X (e.g. that it has the same shape as the internal data).
+        """
+        return self._load()
+
+    def score(self, X, y=None):
```

Comment:
Do you want to provide the full array of scores? or allow the score to be provided as an argument?
https://scikit-learn.org/stable/modules/model_evaluation.html

Reply:
That's a good idea, we could have the scoring metric as an argument - the real reason this is necessary is for visualizer that embed the score in the visualization; for example the `PredictionError` visualizer puts the `R^2` score into the legend; therefore if we didn't have a score method, some visualizers wouldn't work. Perhaps we could open on a follow on story to expand this if necessary?